server/src/main/kotlin/suwayomi/tachidesk/manga/impl/backup/proto/ProtoBackupExport.kt

```diff
@@ -30,11 +34,107 @@ import suwayomi.tachidesk.manga.model.table.MangaStatus
+    private val applicationDirs by DI.global.instance<ApplicationDirs>()
+    private const val lastAutomatedBackupKey = "lastAutomatedBackupKey"
+    private val preferences = Preferences.userNodeForPackage(ProtoBackupExport::class.java)
+
+    private val backupTimer = Timer()
+    private var currentAutomatedBackupTask: TimerTask? = null
+
+    fun scheduleAutomatedBackupTask() {
+        if (!serverConfig.automatedBackups) {
+            currentAutomatedBackupTask?.cancel()
+            return
+        }
+
+        val minInterval = TimeUnit.DAYS.toMillis(1L)
+        val intervalInMS = TimeUnit.DAYS.toMillis(serverConfig.backupInterval.toLong())
+        val backupInterval = intervalInMS.coerceAtLeast(minInterval)
+
+        val lastAutomatedBackup = preferences.getLong(lastAutomatedBackupKey, 0)
+        val initialDelay = backupInterval - (System.currentTimeMillis() - lastAutomatedBackup) % backupInterval
+
+        currentAutomatedBackupTask?.cancel()
+        currentAutomatedBackupTask = object : TimerTask() {
+            override fun run() {
+                cleanupAutomatedBackups()
+                createAutomatedBackup()
+                preferences.putLong(lastAutomatedBackupKey, System.currentTimeMillis())
+            }
+        }
+
+        backupTimer.scheduleAtFixedRate(currentAutomatedBackupTask, initialDelay, backupInterval)
+    }
+
+    private fun createAutomatedBackup() {
+        logger.info { "Creating automated backup..." }
+
+        val inputStream = createBackup(
+            BackupFlags(
+                includeManga = true,
+                includeCategories = true,
+                includeChapters = true,
+                includeTracking = true,
+                includeHistory = true
+            )
+        )
+
+        val automatedBackupDir = File(applicationDirs.automatedBackupRoot)
+        automatedBackupDir.mkdirs()
+
+        val backupFile = File(applicationDirs.automatedBackupRoot, getBackupFilename())
+
+        inputStream.use { input -> backupFile.outputStream().use { output -> input.copyTo(output) } }
+    }
+
+    private fun cleanupAutomatedBackups() {
+        logger.debug { "Cleanup automated backups" }
+
+        val automatedBackupDir = File(applicationDirs.automatedBackupRoot)
+        if (!automatedBackupDir.isDirectory) {
+            return
+        }
+
+        automatedBackupDir.walkTopDown().forEach { file ->
+            try {
+                cleanupAutomatedBackupFile(file)
+            } catch (e: Exception) {
+                // ignore, will be retried on next cleanup
+            }
+        }
+    }
+
+    private fun cleanupAutomatedBackupFile(file: File) {
+        if (!file.isFile) {
+            return
+        }
+
+        val lastAccessTime = file.lastModified()
+        val isTTLReached = System.currentTimeMillis() - lastAccessTime >= TimeUnit.DAYS.toMillis(serverConfig.backupTTL.toLong())
+        if (isTTLReached) {
+            file.delete()
+        }
```

Comment:
More Kotlin Duration
```suggestion
        val lastAccessTime = file.lastModified().milliseconds
        val isTTLReached = System.currentTimeMillis().milliseconds - lastAccessTime >= serverConfig.backupTTL.days
        if (isTTLReached) {
            file.delete()
        }
```